### PR TITLE
Add Lists API

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

> Add support for Lists API

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> None

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,21 @@
...
+  - `GET /lists` with `courier.GetLists` method
+  - `GET /list/{list_id}` with `courier.GetList` method
+  - `PUT /list/{list_id}` with `courier.PutList` method
+  - `DELETE /list/{list_id}` with `courier.DeleteList` method
+  - `PUT /list/{list_id}/restore` with `courier.RestoreList` method
+  - `GET /list/{list_id}/subscriptions` with `courier.GetListSubscriptions` method
+  - `PUT /list/{list_id}/subscriptions` with `courier.PutListSubscriptions` method
+  - `POST /list/{list_id}/subscriptions` with `courier.PostListSubscriptions` method
+  - `PUT /lists/{list_id}/subscriptions/{recipient_id}` with `courier.PutListSubscriptionsRecipient` method
+  - `DELETE /lists/{list_id}/subscriptions/{recipient_id}` with `courier.DeleteListSubscriptionsRecipient` method
```
Comment (aydrian) [inline CHANGELOG.md:18]: For these last 2 we have been using `subscribe` and `unsubscribe` language. So maybe these should be `courier.SubscribeList` and `courier.UnsubscribeList`? Or `courier.ListSubscribe` and `courier.ListUnsubscribe`.

**Review by aydrian — CHANGES_REQUESTED** [review]
Other than the naming, nothing else stands out.

File: CHANGELOG.md (hunk 1 above)
Comment (tk26) [reply CHANGELOG.md:18]: I like that

Comment (tk26) [reply CHANGELOG.md:18]: Going with `ListSubscribe` and `ListSubscribe`

**Review by aydrian — APPROVED** [review]

**Pull request merged**